### PR TITLE
Tiny tweaks based on QE feedback for Datasource guide

### PR DESCRIPTION
**Pull request by MichalMaler**
Based on the QE feedback for the Datasource guide, I am proposing tiny tweaks in the guide and its relevant Javadocs. These need to be backported to 3.2.

The QE request was to add default values or specify if the feature is disabled by default.
Quoting : _Properties `quarkus-agroal_quarkus.datasource."datasource-name".jdbc.leak-detection-interval` and `quarkus.datasource."datasource-name".jdbc.max-lifetime` do not have default duration._

I updated the Javadocs with the @ConfigDocDefault annotation on relevant methods after an SME consultation with @yrodiere , who shared with me the Agroal defaults we, in some cases, share or replace in Quarkus.

```
    Duration leakTimeout = ZERO;       ->      `@ConfigDocDefault("This feature is disabled by default.")`
    Duration validationTimeout = ZERO;      (backgroundValidationInterval)
    Duration reapTimeout = ZERO;               (idleRemovalInterval)
    Duration maxLifetime = ZERO;       ->     `@ConfigDocDefault("This feature is disabled by default.")`
```

Additionally, I updated some links here and there.

### CODE REVIEW

File: extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java [hunk 1]
```diff
@@ -27,8 +27,8 @@ public interface DataSourceReactiveRuntimeConfig {
      * The datasource URLs.
      * <p>
      * If multiple values are set, this datasource will create a pool with a list of servers instead of a single server.
-     * The pool uses a round-robin load balancing when a connection is created to select different servers.
-     * Note: some driver may not support multiple values here.
+     * The pool uses a round-robin load balancing to select different servers when a connection is created.
+     * Note that some drivers might not support multiple values here.
```
Comment (jmartisk) [inline extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/DataSourceReactiveRuntimeConfig.java:31]: Not an English native speaker, but IMO "may not" fits better in this context

File: extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceJdbcRuntimeConfig.java [hunk 2]
```diff
@@ -72,6 +74,7 @@ public interface DataSourceJdbcRuntimeConfig {
     /**
      * The max lifetime of a connection.
      */
+    @ConfigDocDefault("This feature is disabled by default.")
```
Comment (jmartisk) [inline extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceJdbcRuntimeConfig.java:77]: How about "Unlimited by default"?

Comment (MichalMaler) [reply extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceJdbcRuntimeConfig.java]: By default, there is no restriction on the lifespan of a connection. _(much moon, wow)_ :)

**Review by jmartisk — APPROVED** [review]

**Review by yrodiere — APPROVED** [review]

**Comment by MichalMaler** [comment]
Hello @yrodiere @gsmet
Please, do you have an idea what can cause these fails?
image
It is possible to merge this PR if we all agree that my changes can hurt the code?

**Comment by yrodiere** [comment]
> It is possible to merge this PR if we all agree that my changes can hurt the code?

It does look like an unrelated failure.

Though you just pushed again so... let's wait for that run to pass.

**Comment by yrodiere** [comment]
FWIW this looks like the cause:

```
2023-11-14T12:54:56.4011232Z Error response from daemon: Head "https://quay.io/v2/quarkus/ubi-quarkus-mandrel-builder-image/manifests/jdk-21": received unexpected HTTP status: 502 Bad Gateway
```

The build of that module basically crashed, resulting in later steps not being executed, which critically means we never stopped the MariaDB container, triggering cascading failures in following modules since they weren't able to start MariaDB.

@gsmet I wonder if that MariaDB test failure we saw the other day about access rights wasn't a similar (but slightly different) problem: some Maven plugin failed, leading to containers not being stopped, leading to later tests reusing them instead of restarting them (since docker-maven-plugin sometimes does that, I think), leading to the wrong users/roles being present in the database we were testing against.

**Comment by rsvoboda** [comment]
@MichalMaler please prepare dedicated PR for 3.2 branch. Backport can't be applied cleanly, there are conflits in .java file.

Please backport just chages done in docs/src/main/asciidoc/datasource.adoc, that's the safest path for now.

**Comment by MichalMaler** [comment]
> @MichalMaler please prepare dedicated PR for 3.2 branch. Backport can't be applied cleanly, there are conflits in .java file.
> 
> Please backport just chages done in docs/src/main/asciidoc/datasource.adoc, that's the safest path for now.

@rsvoboda Sure thing! It will be done tomorrow. Will add you for the review :)

**Pull request merged**